### PR TITLE
Merge Admin Features

### PR DESCRIPTION
**Pull request by WrinkleYourPizzas**

### CODE REVIEW

File: app/src/main/java/com/example/plannr/DisplayCoursesFragment.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.WRAP_CONTENT);
+
+        LinearLayout page = new LinearLayout((getContext()));
+        page.setOrientation(LinearLayout.VERTICAL);
+        ScrollView scroll = new ScrollView(getContext());
+
+        Button addCourse = new Button(getContext());
+        addCourse.setText("Add Course");
+        addCourse.setTextSize(20);
```
Comment (richardye101) [inline app/src/main/java/com/example/plannr/DisplayCoursesFragment.java:122]: shouldn't this button exist in the layout XML? I think you should access it through `binding.addCourseButton` and have all the properties/attributes set there

File: app/src/main/java/com/example/plannr/DisplayCoursesFragment.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+        addCourse.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                NavHostFragment.findNavController(DisplayCoursesFragment.this)
+                        .navigate(R.id.action_DisplayCoursesFragment_to_adminAddFragment);
+            }
+        });
```
Comment (richardye101) [inline app/src/main/java/com/example/plannr/DisplayCoursesFragment.java:130]: For fragments, i believe code related to buttons or interacting with the view belongs in something like this:
```
public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        // your code goes here

        binding.loginButton.setOnClickListener(view1 -> {
            String email = getEmail();
            String password = getPassword();
            presenter.handleLogin(email, password);
        });
    }
```

File: app/src/main/java/com/example/plannr/LoginViewFragment.java [hunk 3]
```diff
@@ -107,8 +106,11 @@ public void hideLoadingLogin(){
     }
 
     public void loginSuccess(String name){
+//        NavHostFragment.findNavController(LoginViewFragment.this)
+//                .navigate(R.id.action_loginFragment_to_FirstFragment);
         NavHostFragment.findNavController(LoginViewFragment.this)
-                .navigate(R.id.action_loginFragment_to_FirstFragment);
+                .navigate(R.id.action_loginFragment_to_DisplayCoursesFragment);
```
Comment (richardye101) [inline app/src/main/java/com/example/plannr/LoginViewFragment.java:112]: This comment is more for myself; I need to determine where navigation goes depending on the users `isAdmin` boolean value

File: app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditFragment.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import com.example.plannr.databinding.FragmentAdminEditBinding;
+
+/**
+ * AdminEdit fragment class responsible for getting the reference to the components of the view
+ */
+
+public class AdminEditFragment extends Fragment {
+
+    private FragmentAdminEditBinding binding;
+    private AdminEditPresenter presenter2;
```
Comment (richardye101) [inline app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditFragment.java:27]: can we just rename this to `presenter`?

File: app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+                                }
+
+                                Log.i("PREREQUISITE", idPrerequisites);
+
+
+                                //get given course id
+                                String id = "";
+
+                                for(Map.Entry<String, String> set: list.entrySet()){
+                                    if(set.getValue().equals(code)){
```
Comment (richardye101) [inline app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java:108]: what does `set.getValue()` return? in line 95 it seems to be used for comparing the prerequisite courses, but here it seems to be used to compare to the course code?

File: app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+            }
+            return true;
+        }
+        return false;
+    }
+
+    public boolean inputValidator(String courseCode, String courseName, boolean fall, boolean winter, boolean summer, String prerequisites){
+
+        //Initialize warning text reference
+        TextView warningText = view.getEditWarningText();
```
Comment (richardye101) [inline app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java:223]: if we had to conform to MVP for all elements of the app, the setting of text and view elements should live in the view, not the presenter. It should be fine for our purposes however

File: app/src/main/java/com/example/plannr/admin/adminEdit/FirebaseCallback2.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package com.example.plannr.admin.adminEdit;
+
+import java.util.HashMap;
+
+public interface FirebaseCallback2 {
```
Comment (richardye101) [inline app/src/main/java/com/example/plannr/admin/adminEdit/FirebaseCallback2.java:5]: why is this FirebaseCallback2? what happened to the first one?

**Review by richardye101 — COMMENTED** [review]
Overall looks good to me; please confirm this has been tested and if so I can merge and begin linking and wrapping everything together

File: app/src/main/java/com/example/plannr/admin/adminEdit/FirebaseCallback2.java (hunk 7 above)
Comment (KristiDodaj) [reply app/src/main/java/com/example/plannr/admin/adminEdit/FirebaseCallback2.java:5]: FirebaseCallback2 contains a separate function that deals with getting the course selected for editing from the admin landing page. Once a course is selected from the landing page, the course code is pushed to the db under the selection table. Then, the edit fragment will read it from the database. It's essentially a way for the two fragments to communicate as from what I searched up, you can't directly pass information between them.

File: app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java (hunk 5 above)
Comment (KristiDodaj) [reply app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java:108]: The map is composed of ids as keys and course codes as the value. On line 95, we are comparing our prerequisite course codes to the maps/db course codes, and if they match we want to take their id, which is the key of the map at the index where they match. In line 107 we follow the same logic, but now we only want to convert the selected course code for editing into its id form.

File: app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java (hunk 6 above)
Comment (KristiDodaj) [reply app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java:223]: I think the getEditWanrningText method has been defined in the view and is being used in the presenter to get the reference. Either way I don't think we need to strictly abide by it so we are good.

File: app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditFragment.java (hunk 4 above)
Comment (KristiDodaj) [reply app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditFragment.java]: The variable name has now been changed

File: app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java (hunk 6 above)
Comment (richardye101) [reply app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java:223]: I meant to refer to the rest of the code in the function as well, which set the text and colors. Those I believe belong in the view, but yeah nothing that neeeeds change

File: app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java (hunk 5 above)
Comment (richardye101) [reply app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java:108]: Cool! Thanks for explaining

**Review by richardye101 — APPROVED** [review]

File: app/src/main/java/com/example/plannr/DisplayCoursesFragment.java (hunk 2 above)
Comment (KristiDodaj) [reply app/src/main/java/com/example/plannr/DisplayCoursesFragment.java:130]: Essentially, the method is being called from onViewCreated. Matthew structured it in a way that everything is broken down into methods that call each other to simplify everything. Notice that onViewCreated only calls pullData(), and that method uses the commitData() method, which then uses the generatePage() method where the button binding and everything about the design is stored.

**Pull request merged**